thorlcr/graph/thgraphslave.cpp

```diff
@@ -906,15 +906,16 @@ bool CSlaveGraph::recvActivityInitData(size32_t parentExtractSz, const byte *par
             actInitRtnData.clear();
             actInitRtnData.append(true);
             serializeThorException(e, actInitRtnData);
-            e->Release();
-            ret = false;
+            exception.setown(e);
         }
     }
     if (syncInitData() || needActInit)
     {
         if (!queryJobChannel().queryJobComm().send(actInitRtnData, 0, replyTag, LONGTIMEOUT))
             throw MakeStringException(0, "Timeout sending init data back to master");
```

Comment:
And would you throw this or the exception if it was set ?

Reply:
The type of exception is likely moot by the point, if send has failed, then it's likely nothing is going to get back to the master.